src/core/primitives.js

```diff
@@ -81,59 +81,47 @@ var Dict = (function DictClosure() {
     },
 
     // automatically dereferences Ref objects
-    get: function Dict_get(key1, key2, key3) {
-      var value;
-      var xref = this.xref, suppressEncryption = this.suppressEncryption;
-      if (typeof (value = this._map[key1]) !== 'undefined' ||
-          key1 in this._map || typeof key2 === 'undefined') {
-        return xref ? xref.fetchIfRef(value, suppressEncryption) : value;
+    get(key1, key2, key3) {
+      let value = this._map[key1];
+      if (value === undefined && !(key1 in this._map) && key2 !== undefined) {
+        value = this._map[key2];
+        if (value === undefined && !(key2 in this._map) && key3 !== undefined) {
+          value = this._map[key3] || null;
+        }
       }
-      if (typeof (value = this._map[key2]) !== 'undefined' ||
-          key2 in this._map || typeof key3 === 'undefined') {
-        return xref ? xref.fetchIfRef(value, suppressEncryption) : value;
+      if (value instanceof Ref && this.xref) {
+        return this.xref.fetch(value, this.suppressEncryption);
       }
-      value = this._map[key3] || null;
```

Comment:
This, and the identical line in the method below, looks strange to me. Why would you fallback to `null` in only *one* particular case, especially since it's outright wrong if `this._map[key3]` would ever be "falsy" (i.e. `zero` or `false`)!?

There's even a comment in one of the [unit-tests](https://github.com/mozilla/pdf.js/blob/9596d702a993bf3fc8a9a3694bf960ee0b83cbda/test/unit/primitives_spec.js#L71-L72) which mentions this inconsistency, and the only reason that this isn't breaking anything is that there's only *one* [call-site passing in three keys](https://github.com/mozilla/pdf.js/blob/9596d702a993bf3fc8a9a3694bf960ee0b83cbda/src/core/evaluator.js#L2557).

Reply:
Fixed in #11171, so with a rebase this is done.